homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,483 @@
+            if pump_speed == 0:
+                self._state = "off"
+            elif pump_speed == self.coordinator.data[self._item_id].get(
+                "Min-Pump-Speed"
+            ):
+                self._state = "low"
+            elif pump_speed == self.coordinator.data[self._item_id].get(
+                "Max-Pump-Speed"
+            ):
+                self._state = "high"
+        elif pump_type == "SINGLE":
+            if pump_speed == 0:
+                self.state = "off"
+            elif pump_speed == self.coordinator.data[self._item_id].get(
+                "Max-Pump-Speed"
+            ):
+                self._state = "on"
+
+        self._attrs["pump_type"] = pump_type
+
+        return self._state
+
+
+class OmniLogicSaltLevelSensor(OmnilogicSensor):
+    """Define an OmniLogic Salt Level Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        state_key: str,
+        name: str,
+        kind: str,
+        item_id: tuple,
+        device_class: str,
+        icon: str,
+        unit: str,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the salt level sensor."""
+
+        salt_return = self.coordinator.data[self._item_id][self._state_key]
+        unit_of_measurement = self._unit
+
+        if self._unit_type == "Metric":
+            salt_return = round(salt_return / 1000, 2)
+            unit_of_measurement = f"{MASS_GRAMS}/{VOLUME_LITERS}"
+
+        self._state = salt_return
+        self._unit = unit_of_measurement
+
+        return self._state
+
+
+class OmniLogicChlorinatorSensor(OmnilogicSensor):
+    """Define an OmniLogic Chlorinator Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        state_key: str,
+        name: str,
+        kind: str,
+        item_id: tuple,
+        device_class: str,
+        icon: str,
+        unit: str,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the chlorinator sensor."""
+        state = self.coordinator.data[self._item_id][self._state_key]
+        operating_mode = self.coordinator.data[self._item_id].get("operatingMode")
+
+        if operating_mode == "1":
+            self._state = state
+            self._unit = PERCENTAGE
+        elif operating_mode == "2":
```

Comment:
Can the same device shift dynamically between operating modes 1 and 2?

Reply:
Think I know where you are going here. No - it is either variable or on/off. Guessing suggestion is to add this as a guard condition, only add the variable type here in sensors, and save the on/off type for a binary sensor?

Reply:
Yes! :+1:

Reply:
OK have done same for single speed pumps as well since they would be binary on/off.